examples/green_screen/main.cpp

```diff
@@ -0,0 +1,490 @@
+    }
+    if (chessboard_square_length == 0.)
+    {
+        throw std::runtime_error("Chessboard square size is not properly set! Use -s");
+    }
+
+    cout << "Chessboard height: " << chessboard_pattern.height << ". Chessboard width: " << chessboard_pattern.width
+         << ". Chessboard square length: " << chessboard_square_length << endl;
+
+    // Require 2 cameras
+    const int num_devices = k4a::device::get_installed_count();
+    if (num_devices != 2)
+    {
+        throw std::runtime_error("Exactly 2 cameras are required!");
+    }
+
+    // Set up a MultiDeviceCapturer to handle getting many synchronous captures
+    vector<int> device_indices{ 0, 1 };
+    MultiDeviceCapturer capturer(device_indices, color_exposure_usec, powerline_freq);
+
+    // Create configurations. Note that the main MUST be first, and the order of indices in device_indices is not
+    // necessarily preserved because the device_indices may not have ordered according to main and backup
+    vector<k4a_device_configuration_t> configs{ get_main_config(), get_backup_config() };
+    k4a_device_configuration_t &main_config = configs.front();
+    k4a_device_configuration_t &backup_config = configs.back();
+
+    capturer.start_devices(configs);
+
+    // This wraps all the device-to-device details
+    Transformation tr_color_backup_to_color_main =
+        calibrate_devices(capturer, configs, chessboard_pattern, chessboard_square_length);
+
+    k4a::calibration main_calibration = capturer.devices[0].get_calibration(main_config.depth_mode,
+                                                                            main_config.color_resolution);
+    k4a::calibration backup_calibration = capturer.devices[1].get_calibration(backup_config.depth_mode,
+                                                                              backup_config.color_resolution);
+    // Get the transformation from backup depth to backup color using its calibration object
+    Transformation tr_depth_backup_to_color_backup = get_depth_to_color_transformation_from_calibration(
+        backup_calibration);
+
+    // We now have the backup depth to backup color transform. We also have the transformation from the
+    // backup color perspective to the main color perspective from the calibration earlier. Now let's compose the
+    // depth backup -> color backup, color backup -> color main into depth backup -> color main
+    Transformation tr_depth_backup_to_color_main = tr_depth_backup_to_color_backup.compose_with(
+        tr_color_backup_to_color_main);
+
+    // Now, we're going to set up the transformations. DO THIS OUTSIDE OF YOUR MAIN LOOP! Constructing transformations
+    // does a lot of preemptive work to make the transform as fast as possible.
+    k4a::transformation main_depth_to_main_color(main_calibration);
+
+    // We're going to update the existing calibration extrinsics on getting from the backup depth camera to the backup
```

Comment:
>// We're going to update the existing calibration extrinsics [](start = 4, length = 60)

suggestion: essentially you are creating a virtual "calibration" that combine the backup_depth_to_main_color transformation/extrinsics and then this calibration also have the main_color intrinsics, therefore, I like the comment to say "create" instead of "update"

Reply:
I fixed this comment, too- it did used to update the calibration in-place, but it should construct a new one now.